File: compiled_lang/cpp/f30.loop.cpp
// testbox: title="while (true) with break"
#include <iostream>
#include <string>

int main() {
    while (true) {
        std::cout << "Enter your name (quit to exit): ";
        std::string answer;
        std::getline(std::cin, answer);

        if (answer == "quit")
            break;

        std::cout << "Hello " << answer << "!" << std::endl;
    }

    return 0;
}

File: compiled_lang/go/src/f30.loop.go
// testbox: title="bare for {} with break"

package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	stdin := bufio.NewScanner(os.Stdin)
	for {
		fmt.Print("Enter your name (quit to exit): ")
		stdin.Scan()
		answer := stdin.Text()

		if answer == "quit" {
			break
		}

		fmt.Println("Hello " + answer + "!")
	}
}
